frontend/src/app/assignments/submissions/page.tsx

```diff
@@ -1,261 +1,97 @@
 
-  const list = [
-    {
-      title: "Assignment 1",
-      deadline: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      title: "Assignment 2",
-      deadline: new Date(2024, 2, 28, 23, 59, 0),
-    },
-    {
-      title: "Assignment 3",
-      deadline: new Date(2024, 3, 28, 23, 59, 0),
-    },
-    {
-      title: "Assignment 4",
-      deadline: new Date(2024, 4, 28, 23, 59, 0),
-    },
-    {
-      title: "Assignment 5",
-      deadline: new Date(2024, 5, 28, 23, 59, 0),
-    },
-  ];
+  const { user } = useUserContext();
 
-  const submissions = [
-    {
-      id: "1",
-      assignment: "Assignment 1",
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      id: "2",
-      assignment: "Assignment 1",
-      date: new Date(2024, 1, 29, 23, 59, 0),
-    },
-    {
-      id: "3",
-      assignment: "Assignment 2",
-      date: new Date(2024, 2, 28, 23, 59, 0),
-    },
-    {
-      id: "4",
-      assignment: "Assignment 2",
-      date: new Date(2024, 2, 29, 23, 59, 0),
-    },
-    {
-      id: "5",
-      assignment: "Assignment 3",
-      date: new Date(2024, 3, 28, 23, 59, 0),
-    },
-    {
-      id: "6",
-      assignment: "Assignment 4",
-      date: new Date(2024, 4, 28, 23, 59, 0),
-    },
-    {
-      id: "7",
-      assignment: "Assignment 5",
-      date: new Date(2024, 5, 28, 23, 59, 0),
-    },
-  ];
-
-  const studentSubmissions = [
-    {
-      id: "1",
-      assignment: "Assignment 1",
-      name: "Adam Tan",
-      submitted: true,
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      id: "2",
-      assignment: "Assignment 1",
-      name: "Ben Lee",
-      submitted: true,
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      id: "3",
-      assignment: "Assignment 1",
-      name: "Carol Tay",
-      submitted: false,
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      id: "4",
-      assignment: "Assignment 2",
-      name: "Adam Tan",
-      submitted: true,
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      id: "5",
-      assignment: "Assignment 2",
-      name: "Ben Lee",
-      submitted: false,
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-    {
-      id: "6",
-      assignment: "Assignment 3",
-      name: "Carol Tay",
-      submitted: false,
-      date: new Date(2024, 1, 28, 23, 59, 0),
-    },
-  ];
+  const { data: assignments, isLoading } = useQuery({
+    queryKey: ["get-assignments", user],
+    queryFn: async () => {
+      if (!user) {
+        return [];
+      }
+      const submissionsData: Record<string, SubmissionData[]> = {};
+      if (user.role === "student") {
+        // Retrieve all assignments that are published
+        const assignments =  await assignmentService.getAssignmentsByUserId(
+          user.uid,
+          true,
+          true
+        );
+        for (const assignment of assignments) {
```

Comment:
Consider using `Promise.all` to send multiple API calls in parallel if they are not dependent of each other.